scripts/Tools/getTiming

```diff
@@ -1,669 +1,620 @@
-    $found = 0;
-    $min = 0;
-    $max = 0;
-
-    my $strw = $str;
-    $strw =~ s/^\s+//;
-    $strw =~ s/\s+$//;
-    @tmp=`grep -w "$strw" $fin | grep -E '\\(' | grep -v hashtable`;
-    
-    #   print $fh "tcx1 $#tmp $tmp[0]\n";
-    if ($#tmp == 0) {
-	#       print $fh "tcx2 $tmp[0]\n";
-	#pw       if ($tmp[0] =~ m/\s*${strw}\s*\d+\s*(\d*\.\d+)\s*\(.*\)\s*(\d*\.\d+)\s*\(.*\)/) {
-	#pw2 if ($tmp[0] =~ m/\s*${strw}\s*[^\(]+\s+(\d*\.\d+)\s*\(.*\)\s*(\d*\.\d+)\s*\(.*\)/) {
-	if ($tmp[0] =~ m/\s*${strw}\s*\d+\s*\d+\s*\S+\s*\S+\s*(\d*\.\d+)\s*\(.*\)\s*(\d*\.\d+)\s*\(.*\)/) {
-	    #	   print $fh "tcxa $tmp[0]\n";
-	    #           print $fh "tcxb $1 $2 \n";
-	    $max=$1;
-	    $min=$2;
-	    $found=1.0;
-	}
-    }
-    
-    @_[1]=$min;
-    @_[2]=$max;
-    @_[3]=$found;
-    
-}
-
-#----------------------------------------------------------------------------------
-sub gettime2{
-    my($str,$procs,$count,$fin_ref)=@_;
-    my(@tmp,@tmp2);
-    
-#  initialize procs and count
-    $procs = 0;
-    $count = 0;
-    
-    my $strw = $str;
-    $strw =~ s/^\s+//;
-    $strw =~ s/\s+$//;
-    my @tmp = grep( /$strw/, @$fin_ref );
-
-    if ($#tmp == 0) {
-	#pw2 if ($tmp[0] =~ /\s*${strw}\s+([\.\de\+]+)\s+(\d+)\s/) {
-	if ($tmp[0] =~ m/\s*${strw}\s*(\d+)\s*\d+\s*(\S+)/) {
-#pw	    $procs=$2;
-#pw	    $count=$1;
-	    $procs=$1;
-	    $count=$2;
-	}
-	elsif ($tmp[0] =~ /\s*${strw}\s+(\d+)\s/) {
-	    $procs = 1;
-	    $count = $1;
-	}
-    }
-    @_[1]=$procs;
-    @_[2]=$count;
-}
-
-
+#!/usr/bin/env python
+"""
+Get timing information from run
+"""
+
+from standard_script_setup import *
+import argparse, sys, os, logging
+import datetime, shutil, re
+from CIME.case import Case
+
+logger = logging.getLogger(__name__)
+
+def parse_command_line(args, description):
+    parser = argparse.ArgumentParser(
+        usage="\n%s  [-lid|--lid] [-h|--help]" % os.path.basename(args[0]),
+        description=description,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("-lid", "--lid",
+                        help="print using yymmdd-hhmmss format",
+                        default="999999-999999")
+    parser.add_argument("--caseroot", default=os.getcwd(),
+                        help="Case directory to get timing for")
+    args = parser.parse_args(args[1:])
+    CIME.utils.handle_standard_logging_options(args)
+    return args.caseroot, args.lid
+
+
+class getTimingInfo:
+    def __init__(self, name):
+        self.name = name
+        self.tmin = 0
+        self.tmax = 0
+        self.adays = 0
+
+class TimingParser:
+    def __init__(self, caseroot, lid="999999-999999"):
+        self.caseroot = caseroot
+        self.lid = lid
+        self.fin = None
+        self.fout = None
+        self.adays=0
+        self.models = {}
+        for s in ["ATM", "LND", "ROF", "WAV", "ICE", "OCN", "GLC", "CPL"]:
```

Comment:
These generic component names are defined in cime/driver_cpl/cime_config/config_component.xml 
and are available through the CIME.XML.component module method get_value('component")